### PR TITLE
Add Teams Configuration docs page

### PR DESCRIPTION
**Pull request by seesemichaelj**
This PR adds documentation for the changes made in https://github.com/trufflesuite/teams/pull/1241

I also went ahead and added configuration related docs

### CODE REVIEW

File: src/docs/teams/reference/configuration.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+title: Teams | Configuration
+layout: docs.hbs
+---
+# Configuration
+
+## Repository Structure
```
Comment (DiscRiskandBisque) [inline src/docs/teams/reference/configuration.md:7]: This part could be in the Adding Repositories section. I think that's where I'd look for it initially. However, if you planned on expanding this section in the future to contain configurable structures, then it's fine here.

File: src/docs/teams/reference/configuration.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Environment Variables & Secrets
+
+Truffle Teams currently doesn't support injecting environment variables and/or secrets into the build or deployment environments. For Builds, checkout [External CI](/docs/teams/testing/external-ci) which can enable you use an external CI provider that does have those features with Truffle Teams.
+
+## Truffle Config File
+
+Truffle Teams makes some assumptions and modifications to your `truffle-config.js` (or `truffle.js`) configuration file.
+
+#### `networks` Object
```
Comment (DiscRiskandBisque) [inline src/docs/teams/reference/configuration.md:24]: This is super helpful!

File: src/docs/teams/reference/configuration.md (hunk 1 above)
Comment (seesemichaelj) [reply src/docs/teams/reference/configuration.md:7]: Ya the intent was for the latter scenario. It doesn't hurt to have redundant info though

**Comment by seesemichaelj** [comment]
Which image links? I didn't change/add any references to image links

**Pull request merged**